Added a interface and option for the ephemeral type records

RameshIyyar opened the pull request.

RameshIyyar committed:
libguard: Added type string value for the GARD_Sticky_deconfig

Signed-off-by: Ramesh Iyyar <rameshi1@in.ibm.com>
Change-Id: I081ba805dc53a6d259af536667d8e3b6dfebe066

RameshIyyar committed:
libguard: Added a "isEphemeralType" interface

- Added a new "isEphemeralType" interface in the libguard to say
  whether given the guard record type is ephemeral or not.

- It helps the libguard user to check whether the retrieved guard record
  is an ephemeral or persistent type and also, we can avoid caller to
  explicitly add the ephemeral type to filter out if some new type is
  added in the future.

Signed-off-by: Ramesh Iyyar <rameshi1@in.ibm.com>
Change-Id: I36e3d5e2a461f3e9f02d73c6db240f74132ea53e

RameshIyyar committed:
libguard: Use "isEphemeralType" interface instead of custom check

- The libguard supports an API to say whether the give guard type
  is ephemeral or not so using that API instead of adding explicitly.

Signed-off-by: Ramesh Iyyar <rameshi1@in.ibm.com>
Change-Id: Iae4a8f366a00a39e67d2a642e65a92b9d9f6731a

RameshIyyar committed:
guard: Added helper function to print header and record

- Split the "guardList" function to print the header and record so it
  can be used for any guard tool option that's needed to print records.

Tested:

```
> guard -l
ID       | ERROR    |  Type  | Path
00000001 | 00000000 | manual | physical:sys-0/node-0/tpm-0
00000003 | 00000000 | manual | physical:sys-0/node-0/proc-0/mc-0/mi-0/mcc-0/omi-1

> guard -a
ID       | ERROR    |  Type  | Path
ffffffff | 00000000 | manual | physical:sys-0/node-0/tpm-0
```

Signed-off-by: Ramesh Iyyar <rameshi1@in.ibm.com>
Change-Id: I56ab1e66f1e86d519b77efcb0f62cee9c7d4ea95

RameshIyyar committed:
guard: Added hidden option to display ephemeral records

- Added a hidden option to list out the ephemeral type unresolved and
  resolved records.

- The ephemeral type records are intended for the internal purpose
  between the Host and BMC, not to the end-user so added a hidden
  option that helps during debugging some issues.

Tested:
```
> guard -h
Guard Tool
Usage: guard [OPTIONS]

Options:
  -h,--help                   Guard CLI tool options
  -c,--create TEXT            Create Guard record, expects physical path as input
  -i,--invalidate FLOAT       Invalidate a single Guard record, expects record id as input
  -I,--invalidate-all         Invalidates all the Guard records
  -l,--list                   List all the Guard'ed resources
  -r,--reset                  Erase all the Guard records
  -v,--version                Version of GUARD tool

> guard -e
ID       | ERROR    |  Type  | Path
ffffffff | 00000000 | reconfig | physical:sys-0/node-0/dimm-0
ffffffff | 00000000 | sticky_deconfig | physical:sys-0/node-0/ocmb_chip-0
00000001 | 00000000 | reconfig | physical:sys-0/node-0/dimm-0
```

Signed-off-by: Ramesh Iyyar <rameshi1@in.ibm.com>
Change-Id: I5fa25a5e11416f097d94a88db4da2f75e9a654d4

chiragsibm reviewed (approved):
Correct this "- Added a hidden to list out the ephemeral type unresolved and resolved
  records."

RameshIyyar committed:
libguard: Return "unknown" with guard type instead of "nullopt"

- Currently, the "guardReasonToStr" interface returning "nullopt" if
  not able to find the given guard type from the predefined list
  so changed that to return "unknown" with the given guard type for
  the better debug.

Signed-off-by: Ramesh Iyyar <rameshi1@in.ibm.com>
Change-Id: I8d29c916b509ca8ec1ee646bf44e4ce37a19a978

RameshIyyar commented:
> Correct this "- Added a hidden to list out the ephemeral type unresolved and resolved records."

Updated and also, added one more new commit, please review
